app/initializers/setup.js

```diff
@@ -0,0 +1,46 @@
+import { typeOf } from '@ember/utils';
+import config from 'ember-inspector/config/environment';
+import Port from "ember-inspector/port";
+import PromiseAssembler from "ember-inspector/libs/promise-assembler";
+
+export default {
+  name: 'setup',
+  initialize(instance) {
+    // {{EMBER_DIST}} is replaced by the build process.
+    instance.adapter = '{{EMBER_DIST}}';
+
+    // register and inject adapter
+    let Adapter;
+    if (typeOf(instance.adapter) === 'string') {
+      Adapter = instance.resolveRegistration(`adapter:${instance.adapter}`);
+    } else {
+      Adapter = instance.adapter;
+    }
+    register(instance, 'adapter:main', Adapter);
```

Comment:
I know these manual registrations are not new, but I’m curious if we actually need them? I would think that putting a file at `app/adapters/main.js` (Or `app/adapter.js`) would “just work”?

Reply:
Good point. I think these should be turned into services, though it's probably not as simple as putting a file in the correct place since the `adapter` is registered as a different class depending on the environment (Chrome/FF/bookmarklet/tests).

Reply:
Are these adapters like normal adapters for ember-data in an application? If so, I would think they would be automatically registered.

Reply:
I'm going to merge the PR for now as we need this fix to release 3.0, we can look into refactoring this later.